src/Sparklines.js

```diff
@@ -56,7 +56,8 @@ class Sparklines extends React.Component {
         return (
             <svg {...svgOpts} >
                 {React.Children.map(this.props.children, child =>
-                    React.cloneElement(child, { points, width, height, margin })
+                    React.cloneElement(child, { points, limit, width, height, margin, max: typeof max == "undefined" ? Math.max.apply(Math, data) : max,
```

Comment:
Look at this:
```
const data = type == 'custom' ? [value] : dataProcessing[type](ypoints);

const y = dataToPoints({ data, limit, width, height, margin, max, min })[0].y;
```
here dataToPoints gets already transformed data (just one point in many cases), so it has no idea about original data array, and it **can't** calculate correct min and max.
SparklinesReferenceLine, in other side, don't gets original data too. So, we have that min and max of original data array should be calculated in top level component.